version: '3.9'

services:
  db:
    image: postgres:15
    container_name: postgres
    restart: unless-stopped
    environment:
      POSTGRES_DB: projeto3
      POSTGRES_USER: user
      POSTGRES_PASSWORD: password
    volumes:
      - postgres_data:/var/lib/postgresql/data
    ports:
      - '5432:5432'
    healthcheck:
      test: ['CMD-SHELL', 'pg_isready -U user -d projeto3 || exit 1']
      interval: 5s
      timeout: 5s
      retries: 12
      start_period: 10s

  projeto3:
    build:
      context: .
      dockerfile: dockerfile
    container_name: projeto3
    restart: unless-stopped
    working_dir: /app
    command: uvicorn api.app:app --host 0.0.0.0 --port 8000 --reload
    volumes:
      - ./src:/app
    environment:
      DATABASE_URL: postgresql://user:password@db:5432/projeto3
      PYTHONUNBUFFERED: '1'
    ports:
      - '8000:8000'
    depends_on:
      db:
        condition: service_healthy

  pgadministrator:
    image: dpage/pgadmin4
    container_name: pgadministrator
    restart: unless-stopped
    environment:
      PGADMIN_DEFAULT_EMAIL: admin@projeto3.local
      PGADMIN_DEFAULT_PASSWORD: admin
      PGADMIN_CONFIG_ALLOW_SPECIAL_EMAIL_DOMAINS: '["projeto3.local","local","localhost"]'
      PGADMIN_CONFIG_CHECK_EMAIL_DELIVERABILITY: 'False'
    volumes:
      - pgadmin_data:/var/lib/pgadmin
    ports:
      - '5050:80'
    depends_on:
      - db

  grafana:
    image: grafana/grafana
    container_name: grafana
    restart: unless-stopped
    user: '472:472' # evita problema de permissão no volume
    ports:
      - '3050:3000' # << aqui está o ajuste
    volumes:
      - grafana_data:/var/lib/grafana
    environment:
      GF_PATHS_DATA: /var/lib/grafana # ok para SQLite

volumes:
  postgres_data:
  pgadmin_data:
  grafana_data:
